apis/logging/v1/clusterlogging_types.go

```diff
@@ -96,11 +96,20 @@ type ClusterLoggingStatus struct {
 
 // This is the struct that will contain information pertinent to Log visualization (Kibana)
 type VisualizationSpec struct {
+
 	// The type of Visualization to configure
+	// +kubebuilder:validation:Enum=ocp-console;kibana
```

Comment:
I think you also need to allow empty string as a value to not break existing resources.

Reply:
What about people that do not want either of the visualizations? No Kibana for ES and no OCP-console for Loki, because they want to use Loki and their own visualization.

Reply:
@alanconway your call here.  I don't know we have this requirement and it would be a change in behavior from what exists in 5.6.   This may be a better topic or change for "logging 6.0"

Reply:
The whole visualization spec is optional, so people can just omit it. 
If it is present then the type is mandatory so I think this is ok.